subsys/net/lib/aws_fota/src/aws_fota_json.c

```diff
@@ -0,0 +1,192 @@
+	JSON_OBJ_DESCR_PRIM(struct location_obj, host, JSON_TOK_STRING),
+	JSON_OBJ_DESCR_PRIM(struct location_obj, path, JSON_TOK_STRING),
+};
+
+static const struct json_obj_descr job_document_obj_descr[] = {
+	JSON_OBJ_DESCR_PRIM(struct job_document_obj,
+			    operation,
+			    JSON_TOK_STRING),
+	JSON_OBJ_DESCR_PRIM_NAMED(struct job_document_obj,
+				  "fwversion",
+				  fw_version,
+				  JSON_TOK_STRING),
+	JSON_OBJ_DESCR_PRIM(struct job_document_obj, size, JSON_TOK_NUMBER),
+	JSON_OBJ_DESCR_OBJECT(struct job_document_obj,
+			      location,
+			      location_obj_descr),
+};
+
+static const struct json_obj_descr execution_obj_descr[] = {
+	JSON_OBJ_DESCR_PRIM_NAMED(struct execution_obj,
+				  "jobId",
+				  job_id,
+				  JSON_TOK_STRING),
+	JSON_OBJ_DESCR_PRIM(struct execution_obj, status, JSON_TOK_STRING),
+	JSON_OBJ_DESCR_PRIM_NAMED(struct execution_obj,
+				  "queuedAt",
+				  queued_at,
+				  JSON_TOK_NUMBER),
+	JSON_OBJ_DESCR_PRIM_NAMED(struct execution_obj,
+				  "lastUpdatedAt",
+				  last_update_at,
+				  JSON_TOK_NUMBER),
+	JSON_OBJ_DESCR_PRIM_NAMED(struct execution_obj,
+				  "versionNumber",
+				  version_number,
+				  JSON_TOK_NUMBER),
+	JSON_OBJ_DESCR_PRIM_NAMED(struct execution_obj,
+				  "executionNumber",
+				  execution_number,
+				  JSON_TOK_NUMBER),
+	JSON_OBJ_DESCR_OBJECT_NAMED(struct execution_obj,
+				    "jobDocument",
+				    job_document,
+				    job_document_obj_descr),
+};
+
+static const struct json_obj_descr notify_next_obj_descr[] = {
+	JSON_OBJ_DESCR_PRIM(struct notify_next_obj, timestamp, JSON_TOK_NUMBER),
+	JSON_OBJ_DESCR_OBJECT(struct notify_next_obj,
+			      execution,
+			      execution_obj_descr),
+};
+
+static const struct json_obj_descr status_details_obj_descr[] = {
+	JSON_OBJ_DESCR_PRIM_NAMED(struct status_details_obj,
+				  "nextState",
+				  next_state,
+				  JSON_TOK_STRING),
+};
+
+struct update_rsp_obj {
+	const char *status;
+	struct status_details_obj status_details;
+	const char *expected_version;
+	int timestamp;
+	const char *client_token;
+};
+
+static const struct json_obj_descr update_job_exec_stat_rsp_descr[] = {
+	JSON_OBJ_DESCR_PRIM_NAMED(struct update_rsp_obj, "status",
+			status, JSON_TOK_STRING),
+
+	JSON_OBJ_DESCR_OBJECT_NAMED(struct update_rsp_obj,
+				    "statusDetails",
+				    status_details,
+				    status_details_obj_descr),
+
+	JSON_OBJ_DESCR_PRIM_NAMED(struct update_rsp_obj, "expectedVersion",
+			expected_version, JSON_TOK_STRING),
+
+	JSON_OBJ_DESCR_PRIM(struct update_rsp_obj, timestamp, JSON_TOK_NUMBER),
+
+	JSON_OBJ_DESCR_PRIM_NAMED(struct update_rsp_obj, "clientToken",
+			client_token, JSON_TOK_STRING),
+};
+
+int aws_fota_parse_update_job_exec_state_rsp(char *update_rsp_document,
+		size_t payload_len, char *status)
+{
+	struct update_rsp_obj rsp;
+
+	int ret = json_obj_parse(update_rsp_document, payload_len,
+			update_job_exec_stat_rsp_descr,
+			ARRAY_SIZE(update_job_exec_stat_rsp_descr), &rsp);
+
+	if(ret & 1<<0) {
```

Comment:
Spaces needed around the `<<` operator. Or just use `0x01`.

Reply:
done